sql/plan/recursive_cte.go

```diff
@@ -0,0 +1,354 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	sqlerrors "gopkg.in/src-d/go-errors.v1"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+const cteRecursionLimit = 20
```

Comment:
Seems artificially low, any reason not to just let people shoot themselves here?

Reply:
I'm thinking like 1000 at least